lib/i18n-spec/interpolation_cheker.rb

```diff
@@ -0,0 +1,41 @@
+module I18nSpec
+  module InterpolationChecker
+    class << self
+
+      def check(t1, t2)
+        k1, k2 = list_interpolation_keys(t1), list_interpolation_keys(t2)
+        missing_keys = {}
+
+        k1.each do |(scope, keys)|
+          next unless k2.respond_to? :[]
+          diff = keys - (k2[scope] || [])
+          missing_keys[scope] = diff unless diff.empty?
+        end
+
+        missing_keys
+      end
+
+      def list_interpolation_keys(t, scopes = [], result = {})
+        if t.is_a? Hash
+          t.each do |(k, v)|
+            list_interpolation_keys v, [*scopes, k], result
+          end
+        elsif t.is_a? Array
+          t.each_with_index do |v, i|
+            list_interpolation_keys v, [*scopes, i], result
+          end
+        else
+          keys = extract_interpolation_keys t
+          result[scopes.join('.')] = keys unless keys.empty?
+        end
+
+        result
+      end
+
+      def extract_interpolation_keys(translation)
+        translation.to_s.scan(/(?<!%)%\{([^\}]+)\}/).flatten
```

Comment:
Is this a reimplementation of [I18n's existing `INTERPOLATION_PATTERN` constant](https://github.com/svenfuchs/i18n/blob/d4fb998394a389548e91d5f61af3ab77dd03472a/lib/i18n/interpolate/ruby.rb#L5-L9)?

```ruby
INTERPOLATION_PATTERN = Regexp.union(
  /%%/,
  /%\{(\w+)\}/,                               # matches placeholders like "%{foo}"
  /%<(\w+)>(.*?\d*\.?\d*[bBdiouxXeEfgGcps])/  # matches placeholders like "%<foo>.d"
)
```

Is it possible to reuse that here?